quiche/include/quiche.h

```diff
@@ -678,6 +678,54 @@ ssize_t quiche_conn_send_ack_eliciting_on_path(quiche_conn *conn,
                            const struct sockaddr *local, size_t local_len,
                            const struct sockaddr *peer, size_t peer_len);
 
+enum quiche_path_event_type {
+    QUICHE_PATH_EVENT_NEW,
+    QUICHE_PATH_EVENT_VALIDATED,
+    QUICHE_PATH_EVENT_FAILED_VALIDATION,
+    QUICHE_PATH_EVENT_CLOSED,
+    QUICHE_PATH_EVENT_REUSED_SOURCE_CONNECTION_ID,
+    QUICHE_PATH_EVENT_PEER_MIGRATED,
+};
+
+typedef struct quiche_path_event quiche_path_event;
+
+// Retrieves the next event. Returns true if there was an event and ev is et, false otherwise.
+bool quiche_conn_path_event_next(quiche_conn *conn, quiche_path_event **ev);
```

Comment:
@ghedo I wonder if it would be better to change it to:

```
quiche_path_event* quiche_conn_path_event_next(quiche_conn *conn);
```

The caller should then just check if it returned `NULL` or not.

Reply:
actually let me just do this.